### ISSUE TITLE
Weird Self-reflection 

### ISSUE THREAD

**Issue by dorsa01** [dilevin/computer-graphics-ray-tracing#68]
The output for my sphere-and-plane looks like as follows, and it has some self-reflections and weird shadowing at some parts, and some part of the sphere is very bright.
I added the fudge factors for reflection and shadow rays. 
When I set the max_num_reflections to 1, I get the second image where still some part of the sphere is bright.
Where may the problem be?
Thank you
![rgb (7)](https://github.com/user-attachments/assets/fefb69bd-7d6d-47a7-aa1f-5347b96b09a3)
![rgb (8)](https://github.com/user-attachments/assets/0df1f1d8-08a2-440f-b856-21b4abd7e123)

**Comment by vpnda** [comment]
Have you tried https://github.com/dilevin/computer-graphics-ray-tracing/issues/52?

**Comment by dorsa01** [comment]
yes, I checked it and min_t has been set correctly. Anywhere else in the code that you think I need to check?

Thank you

**Comment by vpnda** [comment]
I would double check your implementation of ray reflection as it seems that the reflecting ray might is probably not pointing in the right direction.

**Comment by dorsa01** [comment]
I checked my reflections and I am following the tutorial slides. 
Also I get the same output as ambient-diffuse-specular-shadows.png when I set the max recursion to 0.
Any other suggestion?
